### PR TITLE
[DEVTOOLS-295] Using black check from reusable worflows

### PR DESCRIPTION
**Pull request by rodrigorbl**
Moving Black to be used in the reusable workflow.

### CODE REVIEW

File: configs/.black [hunk 1]
```diff
@@ -0,0 +1,5 @@
+@@ -0,0 +1,4 @@
+[tool.black]
+line-length = 120
```
Comment (JamesHutchisonCarta) [inline configs/.black:3]: This isn't going to be portable. 120 is excessively long and for many the default of 88 (I think?) is better. Black should be soft-wrapping  before then.

Comment (JamesHutchisonCarta) [reply configs/.black:3]: Another way of putting it:
- 88 - soft wrap limit (black will wrap, if it can)
- 120 - hard wrap limit (flake8 will complain)

Comment (rodrigorbl) [reply configs/.black:3]: `carta-web` is being the base to setup the defaults for other codebases and the limit is 120 characters. This limit is also configured for `flake8` and `isort` (on automated-refactoring codebase). I can set to the default here but it will required that we change the defaults on `carta-web`, for example.

Comment (JamesHutchisonCarta) [reply configs/.black:3]: I think it should use the existing black configuration found in the repo. If it's not found, this can use the default black behavior. In either case, we don't need to ship a config file with this.

In general, waiting for the CI to bark at you about something like formatting is a bad experience. I don't think this should be the exclusive thing people are using, which would imply someone is always going to have a config defined somewhere for using locally. Black is really handy for not having to mess with the formatting. Ensuring formatting is standardized is itself a low concern.

Having 120 for isort is fine, because 120 as a hard limit is pretty agreeable and we don't particularly care about how imports look, we just want them to be consistent so you're not doing unnecessary diffs.

File: .github/workflows/linters.yml [hunk 2]
```diff
@@ -167,6 +167,53 @@ jobs:
...
+        with:
+          path: /home/runner/.cache/pip/http
+          key: pip-http-v0.1.0
+        if: steps.cache-venv.outputs.cache-hit != 'true'
+      - uses: actions/cache@v2
+        id: cache-pip-wheel
+        with:
+          path: /home/runner/.cache/pip/wheels
+          key: pip-wheel-v0.1.0
+        if: steps.cache-venv.outputs.cache-hit != 'true'
```
Comment (jimmylai) [inline .github/workflows/linters.yml:196]: Those steps don't seem needed. They referts to step.cache-venv but the step doesn't exist.
We can also remove them from the run_flake8 job.

File: .github/workflows/linters.yml [hunk 3]
```diff
@@ -167,6 +167,53 @@ jobs:
...
+        working-directory: src
+        run: |
+          set -x
+          set -o pipefail
+          . ../fixit-venv/bin/activate
+          
+          if [ ! -f "src/pyproject.toml" ]; then
+            export XDG_CONFIG_HOME=./src/.github/main/configs/.black
+            curl -LJO https://raw.githubusercontent.com/carta/.github/main/configs/.black
+          fi
```
Comment (jimmylai) [inline .github/workflows/linters.yml:208]: We should always use the consistent Black config to apply the same best practice.

File: .github/workflows/linters.yml [hunk 4]
```diff
@@ -167,6 +167,53 @@ jobs:
...
+          set -x
+          set -o pipefail
+          . ../fixit-venv/bin/activate
+          
+          if [ ! -f "src/pyproject.toml" ]; then
+            export XDG_CONFIG_HOME=./src/.github/main/configs/.black
+            curl -LJO https://raw.githubusercontent.com/carta/.github/main/configs/.black
+          fi 
+          
+          mkdir -p logs/
```
Comment (jimmylai) [inline .github/workflows/linters.yml:210]: Why is this needed?

Comment (rodrigorbl) [reply .github/workflows/linters.yml:210]: This path is used to store the logs generated by the `logger` class.

File: .github/workflows/linters.yml (hunk 3 above)
Comment (rodrigorbl) [reply .github/workflows/linters.yml:208]: I keep those local custom setup around to enable existing repositories to reuse the workflows without have to make large changes ([compensation service is an example of it](https://github.com/carta/compensation-service/blob/master/pyproject.toml#L11)). I agree the we should use a consistent config but i think allow custom configs can increase the adoption for existing codebases.

File: .github/workflows/linters.yml [hunk 5]
```diff
@@ -167,6 +155,42 @@ jobs:
...
+        shell: bash
+        working-directory: src
+        run: |
+          set -x
+          set -o pipefail
+          . ../fixit-venv/bin/activate
+          
+          export XDG_CONFIG_HOME=./src/.github/main/configs/.black
+          # curl -LJO https://raw.githubusercontent.com/carta/.github/main/configs/.black
+          curl -LJO https://raw.githubusercontent.com/carta/.github/e393d93ffc24fc743d0facbbbc2e3cd643af2be7/configs/.black
```
Comment (rodrigorbl) [inline .github/workflows/linters.yml:183]: Revert this before merge

File: .github/workflows/linters.yml [hunk 6]
```diff
@@ -134,18 +134,6 @@ jobs:
...
-        with:
-          path: /home/runner/.cache/pip/http
-          key: pip-http-v0.1.0
-        if: steps.cache-venv.outputs.cache-hit != 'true'
-      - uses: actions/cache@v2
-        id: cache-pip-wheel
-        with:
-          path: /home/runner/.cache/pip/wheels
-          key: pip-wheel-v0.1.0
-        if: steps.cache-venv.outputs.cache-hit != 'true'
```
Comment (jimmylai) [inline .github/workflows/linters.yml:148]: Those were removed in the main branch. Rebase to get the latest change.

**Review by jimmylai — APPROVED** [review]
